src/commcare_cloud/environment/secrets/backends/ansible_vault/main.py

```diff
@@ -0,0 +1,147 @@
+import getpass
+import os
+import sys
+from contextlib import contextmanager
+
+import datadog.api
+import yaml
+from ansible.parsing.vault import AnsibleVaultError
+from ansible_vault import Vault
+from memoized import memoized
+from six.moves import shlex_quote
+
+from commcare_cloud.environment.paths import ANSIBLE_DIR
+from commcare_cloud.environment.secrets.secrets_schema import get_known_secret_specs
+
+
+class AnsibleVaultSecretsBackend(object):
```

Comment:
A lot of this class is very specific to ansible vault storage. Is the long term plan to make this a base class with just the public interface or else a composite class and override / delegate the functionality?

Reply:
Yeah the plan is to make a base class with a public interface based on this. There will be a few more renamings of public functions here that are too vault-specific, but the overall structure is mostly right (or at least it's a good start)